src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategy.kt

```diff
@@ -402,7 +402,12 @@ abstract class ManagedStrategy(
 
     private fun isTestThread(iThread: Int) = iThread < nThreads
 
-    private fun inIgnoredSection(iThread: Int): Boolean = !isTestThread(iThread) || ignoredSectionDepth[iThread] > 0
+    /**
+     * Code not in test threads or in ignored sections, which were added by transformer, does not create switch points.
```

Comment:
The execution in an ignored section (added by transformer) or not in a test thread must not add switch points.